### ISSUE TITLE
Tab support

### ISSUE THREAD

**Issue by Sh4dowsand** [GVigne/TheSleepingLion#3]
**Is your feature request related to a problem? Please describe.**
Pressing tab seems to be tabbing through several groups of controls, most of which are rarely used (e.g. the Open AoE editor button). This means you're forced to constantly swap between the keyboard for typing in the text boxes, and the mouse for swapping focus between text boxes.

**Describe the solution you'd like**
Pressing tab while one of the main text boxes are focused should swap focus to the next one, in the order they're shown on the screen. If you're on the final text box ('Custom aliases' on the Class tab, 'ID' on any Card tab), it should either loop back to the first text box on that tab, or possibly move to the next tab.

**Comment by GVigne** [comment]
To be honest, I haven't worked much on the general interface, so there are a lot of things missing (like CTRL+Z not working, tabs can't be moved around….). It will probably be the next big thing I will be working on.
I hadn’t thought of iterating through the text boxes but I can see how useful it would be. As you pointed out, I will probably limit that to CTRL+Tab so you don’t accidentally insert tabs in the text editors (which would change the definition and rendering of the actions you are creating). I guess looping through the same card (ie when pressing CTRL+Tab, if you have the *ID* section in focus, you move back up to the *Card name* section but for the card you are designing) makes a bit more sense – but then I should be adding a shortcut to go to the next card, so you don't have to use the mouse to click on it.

**Comment by Sh4dowsand** [comment]
> there are a lot of things missing (like CTRL+Z not working, tabs can't be moved around….).

I did notice those too, but figured I'd avoid overwhelming you with TOO many tickets in one day xD

Tbf, lack of undo/redo didn't bother me too often, as I plan out my cards in Evernote in plain English, then copy/translate them into Sleeping Lion. Moving tabs shouldn't REALLY matter. It is nice to keep them in level order though... So you could actually do that by having a dropdown sort order instead, with Level, Initiative, Name or File order.

> but then I should be adding a shortcut to go to the next card, so you don't have to use the mouse to click on it.

That would also be nice. Possibly something like Alt + left/right arrows? Or, rather than Ctrl+Tab on ID moving back up to Card Name, it could move to focus the tab itself. Then you can either Ctrl+Tab again to get to Card Name, or use arrow keys to swap tabs?

**Comment by GVigne** [comment]
Having a drop down ordering is neat! Although I might also have to do a tab reordering at some point – I think it’s simpler to order cards thematically, such as having all your Augments side by side.

I’ll have to see what I can do with Gtk, I’m not really familiar with how it focuses on tabs and if there is a nice way to handle shortcuts. Your idea sounds reasonable though: treating the card tab as an other area on which you can focus is pretty neat.

On a side note, I will not have the time to do much programming in the coming week: I will probably start solving issues starting 12th June and hopefully will manage to fix most of them during that week. I’ll probably release a new version once all of this is done – but if you are really eager to get the latest fix, you could always clone the repository to get the new features as soon as I commit changes. Or you could always build your own executable, instructions are in the`/packaging` folder.

Once again, I really appreciate your interest in The Sleeping Lion, and even if I can’t solve many issues this week, do continue to open new ones if you’ve got any request or see anything weird!

**Comment by Sh4dowsand** [comment]
While I am a programmer, I've never really dealt with Python. I'll probably just wait for now, since none of the issues I've hit are blocking problems.

No worries on taking your time though. I'm very much enjoying making classes, so I expect I'll continue using it for quite a while ^^

**Comment by GVigne** [comment]
Tab support has been added in a2000655333eebb664c40f819d965e17ee1112f3.
When a tab has focus, you can use CTRL+TAB to move to the next entry (for example, *Card name* to *Level*). You can move backwards by using  CTRL+ SHIFT + TAB. When reaching the last entry  of the tab, using that shortcut will loop back to the top.

Using Alt + Left Arrow (or Right Arrow) will move the focus to the next (or previous) tab. When reaching the last tab, using this shortcut will loop to the first one (*Class attributes*).

I apologize for the (very) long delay for these fixes. I am still very committed to this project and want to make it grow, but I have had little time recently to work on it. Sorry!
